### ISSUE TITLE
load panthrGraphics into panthrBase

### ISSUE THREAD

**Issue by vanh17** [vanh17/panthrBase#1]

**Comment by vanh17** [comment]
I move the topic in here Dr.  @skiadas

**Comment by vanh17** [comment]
Do you know what does the AMD file looks like? I think it should start with "define" and "somehow" require functions. Thank you Dr. @skiadas
Warmest,

**Comment by skiadas** [comment]
Pretty much all js files in the panthr projects follow an AMD pattern, more specifically one of the UMD patterns (https://github.com/umdjs/umd). 

But you really shouldn't have to create your own AMD/UMD files at this time. What are you trying to do?

**Comment by vanh17** [comment]
I am trying to load panthrGraphics into panthrBase so I can use its graphic tool

**Comment by skiadas** [comment]
Then you should look at how panthrBase is already loading other modules like panthrMath, and how the npm package manager loads and updates packages. Have you done `npm update` in your local panthrBase installation?

**Comment by vanh17** [comment]
I have npm install but have not update it yet. Do you know where the loading reside in the pantrBase?

**Comment by skiadas** [comment]
npm creates a folder called `node_modules` and puts in there all the modules that it has been told to load via the file `package.json`  (https://docs.npmjs.com/files/package.json).

**Comment by vanh17** [comment]
I think it is in panthrBase.js :)))

**Comment by vanh17** [comment]
I did not install npm inside directory of panthrBase. Do I have to do that? My npm_modules now only have browserify, uglify, mocha, and chai
Warmest,

**Comment by skiadas** [comment]
npm is a command line tool that is installed globally. You just need to run it from the folder where your project resides to update the local packages for that project.

https://docs.npmjs.com/

**Comment by vanh17** [comment]
okay, so I put the "panthrGraphics": "*" in the dependency part of the package.json. And I will do npm update in the directory of panthrBase

**Comment by vanh17** [comment]
Okay, I have updated and put the panthrGraphics into the package.json. I copy the tests in panthrGraphics and paste it only in the panthrBase toypage, and try to run it but nothing happen :-? Do you think that means the loading of panthrGraphics into panthrBase was not successful?

**Comment by skiadas** [comment]
You have just included the panthrGraphics files into the `node_modules` folder. You still need to load the module into panthrBase via a "require". You'll have to do something in panthrBase.js

**Comment by vanh17** [comment]
so I put the loader.loadModule(require('./panthrGraphics') into index.js in panthrBase.
Do you think that is the right require? 
Truth to be told, I have not seen any require function that load panthrMath in side directory panthrBase.

**Comment by skiadas** [comment]
You probably want to do something like the `Base.Variable = ... ` line. So maybe `Base.Graphics = require('panthrGraphics')`. You should not put a path specification with it (i.e. the `./`), just the package name. Otherwise it would look for a "local" file instead.

**Comment by vanh17** [comment]
My idea is to reuse the test you implement for panthrGraph to test if we succesfully load it into panthrBase. How can we now use the panthrGraph in our file?

**Comment by skiadas** [comment]
After you rebundle and then load the bundle file on a webpage you should have access to `Base.Graphic` as a global variable. You could then appropriately copy over the parts of the graphics test that you want into a new toypage.

**Comment by vanh17** [comment]
SO I command browserify -o bundle.js. Do you mean that way? and I am kind of confused how can we load the bundle file on a webpage since it is bundle.js. I thought we only can load html or similar file on webpage.

**Comment by skiadas** [comment]
You should be able to do "npm run bundle" from the project's folder. Might have to install browserify and uglifyjs via npm install global, first.

The html page can load script files. Did you look at the code in the html toypages? You will be doing something similar.

**Comment by vanh17** [comment]
I think I just need to run variable.html then, since it has includes bundle.js as a script.

**Comment by skiadas** [comment]
Yes you'll be creating your own file like variable.html

**Comment by vanh17** [comment]
new stuffs take time. But can I use your already implemented tests for graphics? I just do not know what I should put in to the [] after the require. It is now ['../graphic'] but the browser are not happy about it

**Comment by vanh17** [comment]
Basically, I put <script src="bundle.js"></script> in the index.html file in the tests. But it still is not happy about the path inside the require.

**Comment by skiadas** [comment]
Why are you not using the `Base.Graphics = require ...` line I suggested? And I told you not to use a relative path, just the package name on its own. You may need to read up a bit on how node.js's require works.

**Comment by vanh17** [comment]
Yeah, I think I should slow down and read more about node.js and npm too. There are couples new tools here.
